samples/csharp_dotnetcore/21.corebot-app-insights/Startup.cs

```diff
@@ -53,8 +50,31 @@ public void ConfigureServices(IServiceCollection services)
             // Create the Conversation state. (Used by the Dialog system itself.)
             services.AddSingleton<ConversationState>();
 
-            // The Dialog that will be run by the bot.
-            services.AddSingleton<MainDialog>();
+            // Register LUIS recognizer
+            services.AddSingleton<FlightBookingRecognizer>();
+
+            // Register the BookingDialog.
+            services.AddSingleton(sp =>
```

Comment:
I personally prefer the DI way of injecting the IBotTelemetryClient (rather than the lamda)

Reply:
After review, I agree. I have reverted it to pass in the TelemetryClient into MainDialog via DI.  All other dialogs will have it set when they are added to the component dialog.
@WashingtonKayaker Please note this change.  I have also added a comment in MainDialog when dialogs are added to explicitly call out that their TelemetryClient property will be set automatically.